screens/map/ProductDetailsScreen.js

```diff
@@ -13,17 +18,25 @@ class ProductDetailsScreen extends React.Component {
   render() {
     const { currentProduct, store } = this.props.navigation.state.params;
     return (
-      <RowContainer>
-        <ProductInfo product={currentProduct} />
-        <ProductInfoImageContainer>
+      <View>
+        <NavHeaderContainer withMargin backgroundColor="rgba(0,0,0,0)" noShadow>
+          <NavButton onPress={() => this.props.navigation.goBack()}>
+            <FontAwesome5 name="times" solid size={24} />
+          </NavButton>
+          <NavTitle />
+        </NavHeaderContainer>
+        <SpaceBetweenRowContainer
+          style={{
+            marginLeft: 20,
+            marginRight: 8,
+          }}>
+          <ProductInfo style={{ flexShrink: 2 }} product={currentProduct} />
```

Comment:
Didn't find any weird looking things here, but I was testing on iPad so it might be inaccurate.